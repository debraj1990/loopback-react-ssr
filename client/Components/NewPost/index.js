import React, { Component } from 'react'
import axios from 'axios'


class NewPost extends Component{

    state = {
        title: '',
        description: ''
    }

    componentWillMount(){
        if(!localStorage.getItem('__id__')){
            return this.props.history.push('/')
        }
        this.setState({token:localStorage.getItem('__id__')})
    }

    onChange = ({target}) => {
        this.setState({[target.name]:target.value})
    }

    onSubmit = async(e) => {
       e.preventDefault()
        
       try {
          axios.post(`http://localhost:3000/api/Posts?access_token=${this.state.token}`, this.state) 
          return this.props.history.push('/')
       } catch (e) {
           this.setState({error:true})
       }
    }
    render(){
        const { title, description } = this.state
        return (
        <form onSubmit={this.onSubmit} className="col-md-4 col-md-offset-4">
            {
                this.state.error &&
                 <div className="alert alert-danger">
                    <strong>Sorry!</strong> Sorry something went wrong, try again
                 </div> 
            }   
            <div className="form-group">
                <label htmlFor="email">Title:</label>
                <input required onChange={this.onChange} type="text" name='title' value={title} className="form-control" />
            </div>
            <div className="form-group">
            <label htmlFor="description">Description:</label>
                <textarea required onChange={this.onChange} name='description' value={description} className="form-control"></textarea>
            </div>
            <button type="submit" className="btn btn-default">Add Post</button>
        </form>
        )
    }
}

export default NewPost