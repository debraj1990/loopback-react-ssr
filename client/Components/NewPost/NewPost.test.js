import React from 'react';
import {configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import { createMemoryHistory } from 'history'

import NewPost from './'


configure({adapter: new Adapter });


describe('New Post Component', () => {
  
  it('component should push the user if localstorage is not set', () => {
    const history = createMemoryHistory('/')
    const component = shallow(<NewPost history={history} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  });

  it('should render component when localstorage is set', () => {

    localStorage.setItem('__id__', 'dkdkskkekekdkkdkdkdkk')
    const history = createMemoryHistory('/posts/new')
    const component = shallow(<NewPost history={history} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})