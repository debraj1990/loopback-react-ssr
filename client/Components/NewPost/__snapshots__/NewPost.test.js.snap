// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Post Component component should push the user if localstorage is not set 1`] = `
<form
  className="col-md-4 col-md-offset-4"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="email"
    >
      Title:
    </label>
    <input
      className="form-control"
      name="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="description"
    >
      Description:
    </label>
    <textarea
      className="form-control"
      name="description"
      onChange={[Function]}
      required={true}
      value=""
    />
  </div>
  <button
    className="btn btn-default"
    type="submit"
  >
    Add Post
  </button>
</form>
`;

exports[`New Post Component should render component when localstorage is set 1`] = `
<form
  className="col-md-4 col-md-offset-4"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="email"
    >
      Title:
    </label>
    <input
      className="form-control"
      name="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="description"
    >
      Description:
    </label>
    <textarea
      className="form-control"
      name="description"
      onChange={[Function]}
      required={true}
      value=""
    />
  </div>
  <button
    className="btn btn-default"
    type="submit"
  >
    Add Post
  </button>
</form>
`;
